src/pytorch_lightning/tuner/tuning.py

```diff
@@ -129,22 +167,34 @@ def scale_batch_size(
                 - ``model.hparams``
                 - ``trainer.datamodule`` (the datamodule passed to the tune method)
         """
-        self.trainer.auto_scale_batch_size = True
-        result = self.trainer.tune(
-            model,
-            train_dataloaders=train_dataloaders,
-            val_dataloaders=val_dataloaders,
-            datamodule=datamodule,
-            scale_batch_size_kwargs={
-                "mode": mode,
-                "steps_per_trial": steps_per_trial,
-                "init_val": init_val,
-                "max_trials": max_trials,
-                "batch_arg_name": batch_arg_name,
-            },
+        self.trainer.state.fn = TrainerFn.TUNING
+        self.tuning = True
+
+        _check_tuner_configuration(self.trainer, train_dataloaders, val_dataloaders, dataloaders, method)
+
+        batch_size_finder: Callback = BatchSizeFinder(
+            mode=mode,
+            steps_per_trial=steps_per_trial,
+            init_val=init_val,
+            max_trials=max_trials,
+            batch_arg_name=batch_arg_name,
         )
+        # do not continue with the loop in case trainer.tuner is used
+        batch_size_finder._early_exit = True
+        self.trainer.callbacks = [batch_size_finder] + self.trainer.callbacks
+
+        if method == "fit":
+            self.trainer.fit(model, train_dataloaders, val_dataloaders, datamodule)
```

Comment:
This will overwrite the `Tuning` TrainerFn. so might as well remove it

Reply:
in a followup once both the PR get's merged.